modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -80,20 +93,33 @@ private long parseMillis(String date) {
             // This means that we leave the existing parsing logic in place, but will fall back to the new java date parsing logic, if an
             // "8" is prepended to the date format string
             int year = LocalDate.now(ZoneOffset.UTC).getYear();
+            ZoneId zoneId = DateUtils.dateTimeZoneToZoneId(timezone);
             if (format.startsWith("8")) {
                 DateFormatter formatter = DateFormatter.forPattern(format)
                     .withLocale(locale)
-                    .withZone(DateUtils.dateTimeZoneToZoneId(timezone));
+                    .withZone(zoneId);
                 return text -> {
-                    ZonedDateTime defaultZonedDateTime = Instant.EPOCH.atZone(ZoneOffset.UTC).withYear(year);
                     TemporalAccessor accessor = formatter.parse(text);
-                    long millis = DateFormatters.toZonedDateTime(accessor, defaultZonedDateTime).toInstant().toEpochMilli();
+                    // if there is no year, we fall back to the current one and
+                    // fill the rest of the date up with the parsed date
+                    if (accessor.isSupported(ChronoField.YEAR) == false) {
+                        ZonedDateTime newTime = Instant.EPOCH.atZone(ZoneOffset.UTC).withYear(year);
```

Comment:
Shouldn't we use the "current" year in the specified time zone instead of UTC? Wouldn't we otherwise either use the next or the previous year around the turn of the year?

Reply:
I resembled the joda time behaviour here. I also created a test that compares joda time with java time to make sure the output is the same